apps/rush-lib/src/logic/buildCache/AmazonS3BuildCacheProvider.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { Terminal } from '@rushstack/node-core-library';
+
+import { EnvironmentConfiguration, EnvironmentVariableNames } from '../../api/EnvironmentConfiguration';
+import { CloudBuildCacheProviderBase } from './CloudBuildCacheProviderBase';
+import { S3Client, GetObjectCommand, PutObjectCommand, GetObjectCommandOutput } from '@aws-sdk/client-s3';
```

Comment:
Because Rush gets invoked interactively for [tab completion](https://rushjs.io/pages/developer/tab_completion/), the process startup time is fairly important.  We use the [Import.lazy()](https://github.com/microsoft/rushstack/blob/fc09e5d90b77ae1e818fa2d60a13d497a6d9c015/libraries/node-core-library/src/Import.ts#L175) API to manage this.  (Search the Rush code base for examples of how to use it.)  

Note that startup time is an orthogonal issue from NPM installation time (which I mentioned in my other comment).

Reply:
Would it make sense to `Import.lazy` the `AmazonS3BuildCacheProvider` itself rather than its dependencies?

Reply:
I put an `Import.lazy` in front of both build cache providers and got the module count back down to 349, and there seem to be no traces of "aws" nor "azure" in the log now.